lib/conversion/widgets/conversion_screen.dart

```diff
@@ -0,0 +1,234 @@
+                ),
+                SwapIconButton(
+                  onPressed: () {
+                    CryptoEntity temp = widget.asset;
+                    setState(() {
+                      widget.asset = cryptoConverted;
+                      cryptoConverted = temp;
+                      widget.coinAmmount = dp(ref
+                          .read(coinAmmountProvider)[
+                              data.indexOf(widget.asset as CryptoViewData)]
+                          .toString());
+                    });
+                    convertController.clear();
+                    buttonValidation();
+                    convertedValue('0.0');
+                  },
+                ),
+                CoinButton(
+                  data: data,
+                  onPressed: () {
+                    showModalBottomSheetCryptos(
+                      context,
+                      data,
+                      ListView(
+                        children: data.map((crypto) {
+                          return Column(
+                            children: [
+                              const Divider(thickness: 1),
+                              ListTile(
+                                onTap: () {
+                                  CryptoEntity temp = cryptoConverted;
+                                  setState(() {
+                                    cryptoConverted = crypto;
+                                    if (cryptoConverted.id == widget.asset.id) {
+                                      widget.asset = temp;
+                                    }
+                                  });
+                                  buttonValidation();
+                                  convertedValue(convertController.text);
+                                  Navigator.of(context).pop();
+                                },
+                                title: Text(crypto.symbol.toUpperCase()),
+                                subtitle: Text(crypto.name),
+                                trailing:
+                                    const Icon(Icons.keyboard_arrow_right),
+                              ),
+                            ],
+                          );
+                        }).toList(),
+                      ),
+                    );
+                  },
+                  asset: cryptoConverted,
+                ),
+              ],
+            ),
+            CoinTextField(
+              formKey: formKey,
+              controller: convertController,
+              asset: widget.asset,
+              onChanged: (value) {
+                buttonValidation();
+                convertedValue(value);
+              },
+              validator: (value) {
+                if (value == '' || value == null) {
+                  return 'Valor não pode ser igual a 0';
+                } else if (validCoinValue(value)) {
+                  return 'O valor inicial não pode ser um caractere especial';
```

Comment:
Dica: o TextFormField possui um atributo chamado "inputFormatters" que ajuda a formatar o que o usuário está escrevendo dentro do TextForm, dentro dele você pode usar algum pacote ou RegEx, tal como esse daqui:
RegEx(r'^(\d+)?,?\d{0,6}')
Página para teste: https://www.regextester.com/index.php?fam=124969
Onde: (^ -> Começo da linha, \d -> permite todos os numeros (parecido com [0-9]), ,? -> permite o uso de virgula, \d{0,6} -> permite a digitação de 6 números após a virgula)
Obs. como você está usando Decimal, se o usuário digitar "-" como primeiro caracter teu aplicativo vai dar o seguinte erro "Could not parse BigInt", é por isso que a melhor opção seria não permitir o uso deste tipo de caracteres.

Reply:
Feito